R/int_pctl.R

```diff
@@ -0,0 +1,224 @@
+#' given by [collect_metrics()] since a different estimator is used. Since
+#' random numbers are used in sampling, set the random number seed prior to
+#' running this function.
+#'
+#' The number of bootstrap samples should be large to have reliable intervals.
+#' The defaults reflect the fewest samples that should be used.
+#'
+#' The computations for each configuration can be extensive. To increase
+#' computational efficiency parallel processing can be used. The \pkg{foreach}
+#' package is used here. To execute the resampling iterations in parallel,
+#' register a parallel backend function. See the documentation for
+#' [foreach::foreach()] for examples. The `allow_par` argument can be used to
+#' avoid parallelism.
+#'
+#' Also, if a censored regression model used numerous evaluation times, the
+#' computations can take a long time unless the times are filtered with the
+#' `eval_time` argument.
+#' @seealso [rsample::int_pctl()]
+#' @references Davison, A., & Hinkley, D. (1997). _Bootstrap Methods and their
+#'  Application_. Cambridge: Cambridge University Press.
+#'  doi:10.1017/CBO9780511802843
+#' @examplesIf !tune:::is_cran_check() & tune:::should_run_examples("modeldata")
+#' data(Sacramento, package = "modeldata")
+#' library(rsample)
+#' library(parsnip)
+#'
+#' f <- log10(price) ~ beds + baths + sqft + type + latitude + longitude
+#'
+#' set.seed(13)
+#' sac_rs <- vfold_cv(Sacramento)
+#'
+#' lm_res <-
+#'   linear_reg() %>%
+#'   fit_resamples(
+#'     log10(price) ~ beds + baths + sqft + type + latitude + longitude,
+#'     resamples = sac_rs,
+#'     control = control_resamples(save_pred = TRUE)
+#'   )
+#'
+#' set.seed(31)
+#' int_pctl(lm_res)
+#' @export
+int_pctl.tune_results <- function(.data, metrics = NULL, times = 1001,
+                                  eval_time = NULL, parameters = NULL,
+                                  alpha = 0.05, allow_par = TRUE,
+                                  event_level = "first", ...) {
+
+  rlang::check_dots_empty(...)
+
+  # check eval_time and set default when null
+  eval_time <- default_eval_time(eval_time, .data$.metrics[[1]])
+  .data$.predictions <- filter_eval_times(.data, eval_time)
+
+  y_nm <- outcome_names(.data)
+
+  config_keys <- get_configs(.data, parameters = parameters)
+  p <- length(config_keys)
+
+  if (is.null(metrics)) {
+    metrics <- .get_tune_metrics(.data)
+  }
+
+  res <-
+    purrr::map2_dfr(
+      config_keys, sample.int(10000, p),
+      ~ compute_by_config(.x, .y, .data, metrics, times, allow_par, event_level)
+    ) %>%
+    dplyr::arrange(.config, .metric)
+  dplyr::as_tibble(res)
+}
+
+get_int_p_operator <- function(allow = TRUE) {
+  is_par <- foreach::getDoParWorkers() > 1
+  if (allow && is_par) {
+    res <- foreach::`%dopar%`
+  } else {
+    res <- foreach::`%do%`
+  }
+  res
+}
+
+compute_by_config <- function(config, seed, x, metrics, times, allow_par, event_level) {
```

Comment:
Some of these helper functions have names that make sense in the context of this file but are a bit generic when the scope is all of the package. Maybe they could get names that are a little bit more specific?